rewrite-maven/src/main/java/org/openrewrite/maven/tree/Pom.java

```diff
@@ -91,6 +91,9 @@ public static int getModelVersion() {
     @Nullable
     String packaging;
 
+    @Builder.Default
+    Map<String, String> constants = emptyMap();
```

Comment:
I'll explore some more where I can, but I continue to be of the opinion that we should add these properties to the MavenResolutionResult Marker, not the Pom. The tree package classes are there to model what we saw in the source files; the markers can add information pulled in from the environment, like in this case or with active profiles.